File: VB/t329154VB/Controllers/HomeController.vb
Imports System.Globalization

Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        Return View()
    End Function

    <HttpPost>
    Public Function Index(ByVal language As String) As ActionResult
        Session("CurrentCulture") = CultureInfo.GetCultureInfo(language)
        CustomLocalizationHelper.ApplyCurrentCulture()
        Return View()
    End Function

    Public Function GridViewPartial() As ActionResult
        CustomLocalizationHelper.ApplyCurrentCulture()
        Return PartialView(Invoice.GetData())
    End Function
End Class

File: CS/t329154/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using t329154.Models;

namespace t329154.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index()
        {
            return View();
        }

        [HttpPost]
        public ActionResult Index(string language)
        {
            Session["CurrentCulture"] = CultureInfo.GetCultureInfo(language);
            CustomLocalizationHelper.ApplyCurrentCulture();
            return View();
        }

        public ActionResult GridViewPartial()
        {
            CustomLocalizationHelper.ApplyCurrentCulture();
            return PartialView(Invoice.GetData());
        }
    }
}